src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -202,29 +206,80 @@ trait WorkspaceComponent {
     }
 
     def getAuthorizedRealms(workspaceIds: Seq[String], user: RawlsUserRef): ReadAction[Seq[Option[RawlsGroupRef]]] = {
-      DBIO.sequence(workspaceIds.map { id =>
-        findById(id) flatMap {
-          case None => DBIO.successful(None)
-          case Some(workspace) => DBIO.successful(workspace.realm)
-        }
-      }) flatMap { allRealms =>
+      val realmQuery = for {
+        workspace <- workspaceQuery if workspace.id.inSetBind(workspaceIds.map(UUID.fromString))
+      } yield workspace.realmGroupName
+
+      realmQuery.result flatMap { allRealms =>
         val flatRealms = allRealms.flatten.toSet
         DBIO.sequence(flatRealms.toSeq.map { realm =>
-          rawlsGroupQuery.loadGroupIfMember(realm, user) flatMap {
+          val groupRef = RawlsGroupRef(RawlsGroupName(realm))
+          rawlsGroupQuery.loadGroupIfMember(groupRef, user) flatMap {
             case None => DBIO.successful(None)
-            case Some(_) => DBIO.successful(Some(realm))
+            case Some(_) => DBIO.successful(Some(groupRef))
           }
         })
       }
     }
 
-    private def workspaceAttributes(workspaceId: UUID) = for {
-      workspaceAttrRec <- workspaceAttributeQuery if workspaceAttrRec.workspaceId === workspaceId
+    def listAccessGroupMemberEmails(workspaceIds: Seq[UUID], accessLevel: WorkspaceAccessLevel): ReadAction[Map[UUID, Seq[String]]] = {
+      val accessQuery = workspaceAccessQuery.filter(access => access.workspaceId.inSetBind(workspaceIds) && access.isRealmAcl === false && access.accessLevel === accessLevel.toString)
+
+      val subGroupEmailQuery = accessQuery join rawlsGroupQuery.subGroupEmailsQuery on {
+        case (wsAccess, (parentGroupName, subGroupEmail)) => wsAccess.groupName === parentGroupName } map {
+        case (wsAccess, (parentGroupName, subGroupEmail)) => (wsAccess.workspaceId, subGroupEmail) }
+
+      val userEmailQuery = accessQuery join rawlsGroupQuery.userEmailsQuery on {
+        case (wsAccess, (parentGroupName, userEmail)) => wsAccess.groupName === parentGroupName } map {
+        case (wsAccess, (parentGroupName, userEmail)) => (wsAccess.workspaceId, userEmail) }
+
+      (subGroupEmailQuery union userEmailQuery).result.map { wsIdAndEmails =>
+        wsIdAndEmails.groupBy { case (wsId, email) => wsId }.
+          mapValues(_.map { case (wsId, email) => email }) }
+    }
+
+    def listSubmissionSummaryStats(workspaceIds: Seq[UUID]): ReadAction[Map[UUID, WorkspaceSubmissionStats]] = {
+      val workflowDatesQuery = for {
+        submissions <- submissionQuery if submissions.workspaceId.inSetBind(workspaceIds)
+        workflows <- workflowQuery if submissions.id === workflows.submissionId
+      } yield (submissions.workspaceId, workflows.status, workflows.statusLastChangedDate)
+
+      val workflowDatesGroupedQuery = workflowDatesQuery.groupBy { case (wsId, status, _) => (wsId, status) }.
```

Comment:
Add comment with a description of how this transforms the above query.

I believe it is showing us the last time each of the Workflow statuses "Done", "Failed", etc have been seen per Submission, without regard to which Workflow has each status - but it took me some time to work that out.